stacks_queues/queue.py

```diff
@@ -16,47 +16,90 @@ def __init__(self):
         self.rear = -1
         self.size = 0
       
-
+    # put something in back and comes off through the front
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
-
+        # checks to see if the list is empty
+        # if list is empty we want to add the element to the list set it to front
+        if self.front == -1 and self.rear == -1:
+            self.front += 1
+            self.store[self.front] = element
+        
+        elif self.front == (self.rear + 1) % INITIAL_QUEUE_SIZE:
+            raise
+        
+        else:
+            self.rear = (self.rear + 1) % INITIAL_QUEUE_SIZE
+            self.store[self.rear] = element 
+        
+        
+        
     def dequeue(self):
         """ Removes an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
             returns None
         """
-        pass
+        if self.front == -1 and self.rear == -1:
+            return None
+        else:
+            self.store[self.front + 1] = self.front
```

Comment:
There's nothing here as well that might allow the front to wrap around once it reaches the end of `self.store`.